### ISSUE TITLE
Modify plugins to pass `path` instead of `node` into visitor handlers.

### ISSUE THREAD

**Issue by rwjblue** [ember-template-lint/ember-template-lint#186]
The initial implementation is trivial:

```
interface Path {
  node: ASTNode
}
```

But in the long run, it can become much much more advanced.  Some possible functionality that it can expose:

* `path.isBlockParam()`
* All of the helpers in [lib/helpers/ast-node-info.js](https://github.com/rwjblue/ember-template-lint/blob/master/lib/helpers/ast-node-info.js)
* expose `path.parent` (which should hopefully allow much less manual state tracking)

**Comment by rwjblue** [comment]
@mmun - Where did we end up on this?  I vaguely recall that we were going to shoot for something different now?

**Comment by mmun** [comment]
I'm thinking something like 

```ts
class extends Visitor {
  @enter MustacheStatement(node) {
    this.isBlockParam(node.params[0])
    this.isElement(this.getParent(node))
  }
}
```

or if we get the decorators working

```ts
class extends Visitor {
  [enter('MustacheStatement')](node) {
    this.isBlockParam(node.params[0])
    this.isElement(this.getParent(node))
  }
}
```

**Comment by rwjblue** [comment]
👍 - thank you for summarizing nicely!

I think this issue no longer applies as a blocker for 0.7.0 then, removing milestone.

**Comment by rwjblue** [comment]
See https://github.com/glimmerjs/glimmer-vm/pull/985.

**Comment by lifeart** [comment]
Closing as implemented in https://github.com/ember-template-lint/ember-template-lint/commit/0fcfa17180773bec0e436de8f778dbaf5a8570d1
